oauth2client/client.py

```diff
@@ -2056,10 +2048,7 @@ def step2_exchange(self, code=None, http=None, device_flow_info=None):
         if self.user_agent is not None:
             headers['user-agent'] = self.user_agent
 
-        if http is None:
-            http = transport.get_http_object()
```

Comment:
What happens if `http` is `None` here? Just an error? Should we add a test for this?